beacon_node/network/src/sync/manager.rs

```diff
@@ -137,9 +139,20 @@ pub enum BatchProcessResult {
     Failed {
         imported_blocks: bool,
         peer_action: Option<PeerAction>,
+        mode: FailureMode,
     },
 }
 
+/// A wrapper struct that represents which chains are currently waiting on
+/// execution layer to become available.
+///
+/// A value of `true` implies that particular sync mode is currently waiting
+/// for execution to become available.
+struct WaitingOnExecution {
+    range: bool,
+    block_lookup: bool,
+}
```

Comment:
in what scenario would happen that one is stalled and the other is not, beyond the time in which one is aware of it but the other is not?

Reply:
I separated them so that we don't resume sync on `RangeSync` if range sync had not even started and then subsequently stalled. Basically this scenario:
1. Node is fully synced/undergoing historical sync, now execution goes offline
2. `ParentLookup` gets paused on execution
3. Execution comes back online, `ParentLookup` status is set to `Activated` and  `execution_ready` is sent to `RangeSync` even though syncing is stopped. 

Maybe we can handle this better in `RangeSync` such that there's no invalid state if `ChainState == Idle`?